Defra.Cdp.Backend.Api/Program.cs

```diff
@@ -53,8 +56,45 @@
 
 // Add health checks and http client
 builder.Services.AddHealthChecks();
-builder.Services.AddHttpClient();
-builder.Services.AddHttpProxyClient(logger);
+
+builder.Services.AddHttpClient("DefaultClient", client =>
+    {
+        client.DefaultRequestHeaders.Accept.Clear();
+        client.DefaultRequestHeaders.Add(
+            "Accept", "application/json");
+        client.DefaultRequestHeaders.Add("User-Agent",
+            "cdp-portal-backend");
+    })
+    .AddHeaderPropagation();
+
+builder.Services.AddHttpClient("GitHubClient", client =>
+{
+    client.DefaultRequestHeaders.Accept.Clear();
+    client.DefaultRequestHeaders.Add(
+        "Accept", "application/vnd.github+json");
+    client.DefaultRequestHeaders.Add("X-GitHub-Api-Version", "2022-11-28");
+    client.DefaultRequestHeaders.Add("User-Agent",
+        "cdp-portal-backend"); // required by GitHub API or else you get 403
+}).ConfigurePrimaryHttpMessageHandler(() => Proxy.ProxyHttpMessageHandler(logger)).AddHeaderPropagation();
+
+builder.Services.AddHttpClient("proxy", client =>
+{
+    client.DefaultRequestHeaders.Accept.Clear();
+    client.DefaultRequestHeaders.Add(
+        "Accept", "application/json");
+    client.DefaultRequestHeaders.Add("User-Agent",
+        "cdp-portal-backend");
+}).ConfigurePrimaryHttpMessageHandler(() => Proxy.ProxyHttpMessageHandler(logger));
+
+builder.Services.AddHeaderPropagation(options =>
+{
+    var tracingEnabled = builder.Configuration.GetValue<bool>("Tracing:Enabled");
+    var tracingHeader = builder.Configuration.GetValue<string>("Tracing:Header");
+    if (tracingEnabled && string.IsNullOrEmpty(tracingHeader) == false)
+    {
+        options.Headers.Add(tracingHeader);    
+    }
+});
```

Comment:
Possible tidy up option is to move the http client setup code into a static class like
```
public static HttpClients {

    public static IServiceCollection AddGithubClient(this IServiceCollection services)
        {
            services.AddHttpClient("GitHubClient", client =>
            {
                client.DefaultRequestHeaders.Accept.Clear();
                // etc....
            }).ConfigurePrimaryHttpMessageHandler(provider => // etc ...)
            return services;
        }

}
```

then in program.cs we can just do `builder.AddGitHubClient().AddProxyClient()` etc

(or group them any which way, e.g.  `.AddGitubPoller` which adds the client + scheduler etc)